### PR TITLE
ENH: Add annotations to `np.core.multiarray` part 3/4

### PR DESCRIPTION
**Pull request by BvB93**
The third part in a series of PRs for adding annotations to `np.core.multiarray`.

Pretty much all remaining (public) functions in the namespace are now annotated, 
with the exception of `nditer` and `nested_iters`.

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -1280,7 +1280,7 @@
 
 add_newdoc('numpy.core.multiarray', 'fromstring',
     """
-    fromstring(string, dtype=float, count=-1, sep='', *, like=None)
+    fromstring(string, dtype=float, count=-1, *, sep, like=None)
```
Comment (BvB93) [inline numpy/core/_add_newdocs.py:1283]: `fromstring` is currently a bit of an odd one. It does have a default value for `sep`, 
but it is deprecated, this leaving us with a mandatory parameter right in the middle of a function signature, 
As a compromise I now documented it as a mandatory keyword-only parameter.

Any thoughts here?

Comment (charris) [reply numpy/core/_add_newdocs.py:1283]: When was it deprecated?

Comment (BvB93) [reply numpy/core/_add_newdocs.py:1283]: By the looks of it in 1.14:
https://numpy.org/doc/stable/reference/generated/numpy.fromstring.html

Comment (BvB93) [reply numpy/core/_add_newdocs.py:1283]: > Mandatory seems reasonable. Maybe it should get a better default value?

I imagine that `" "` would be the most obvious candidate here.

**Comment by charris** [comment]
Thanks Bas.

**Pull request merged**